src/main/scala/com/lucidworks/spark/example/ml/SVMExampleScala.scala

```diff
@@ -0,0 +1,161 @@
+import org.apache.spark.mllib.feature.HashingTF
+import org.apache.spark.mllib.feature.Normalizer
+import org.apache.spark.mllib.feature.StandardScaler
+import org.apache.spark.mllib.feature.StandardScalerModel
+import scala.collection.JavaConverters._
+import scala.collection
+
+import scala.collection.immutable
+
+object SVMExampleScala {
+  val DEFAULT_NUM_FEATURES = "1000000"
+  val DEFAULT_NUM_ITERATIONS = "200"
+  val DefaultZkHost = "localhost:9983"
+  val DefaultCollection = "twitter_sentiment"
+}
+
+class SVMExampleScala extends SparkApp.RDDProcessor  {
+
+  import SVMExampleScala._
+  def getName = "mllib-svm-scala"
+  def getOptions = Array(
+    Option.builder().longOpt("indexTrainingData").hasArg.required(false).desc(
+      s"Path to training data to index").build(),
+    Option.builder().longOpt("indexTestData").hasArg.required(false).desc(
+      s"Path to test data to index").build(),
+    Option.builder().longOpt("sample").hasArg.required(false).desc(
+      s"Fraction (0 to 1) of full dataset to sample from Solr, default is 1").build(),
+    Option.builder().longOpt("numFeatures").hasArg.required(false).desc(
+      s"Number of features; default is $DEFAULT_NUM_FEATURES").build(),
+    Option.builder().longOpt("numIterations").hasArg.required(false).desc(
+      s"Number of iterations; default is $DEFAULT_NUM_ITERATIONS").build(),
+    Option.builder().longOpt("modelOutput").hasArg.required(false).desc(
+      s"Model output path; default is mllib-svm-sentiment").build()
+  )
+
+  override def run(conf: SparkConf, cli: CommandLine): Int = {
+    val sc = new SparkContext(conf)
+    val sqlContext = new SQLContext(sc)
+    val csvSchema = StructType(StructField("polarity",StringType, true) ::
+      StructField("id",StringType, true) ::
+      StructField("date",StringType, true) ::
+      StructField("query",StringType, true) ::
+      StructField("username",StringType, true) ::
+      StructField("tweet_txt",StringType, true) :: Nil)
+
+    val indexTrainingData = cli.getOptionValue("indexTrainingData")
+    if (indexTrainingData != null) {
+      var csvDF = sqlContext.read.format("com.databricks.spark.csv").schema(csvSchema).option("header", "false").load(indexTrainingData)
+      csvDF = csvDF.repartition(4)
+
+      val options = immutable.HashMap(
+        "zkhost" -> cli.getOptionValue("zkHost", DefaultZkHost),
+        "collection" -> cli.getOptionValue("collection", DefaultCollection),
+        "soft_commit_secs" -> "10")
+      csvDF.write.format("solr").options(options).mode(SaveMode.Overwrite).save()
+    }
+
+    val indexTestData = cli.getOptionValue("indexTestData");
+    if (indexTestData != null) {
+      var csvDF = sqlContext.read.format("com.databricks.spark.csv").schema(csvSchema).option("header", "false").load(indexTestData)
+      csvDF = csvDF.withColumnRenamed("polarity", "test_polarity")
+
+      val options = immutable.HashMap(
+        "zkhost" -> cli.getOptionValue("zkHost", DefaultZkHost),
```

Comment:
We can avoid this redundancy by reading in the option values in the beginning and re-use them wherever needed.